data/trie/extensionNode.go

```diff
@@ -378,3 +378,30 @@ func (en *extensionNode) print(writer io.Writer, index int) {
 	_, _ = fmt.Fprint(writer, str)
 	en.child.print(writer, index+len(str))
 }
+
+func (en *extensionNode) deepClone() node {
```

Comment:
You should check if en.isEmptyOrNil(), as all the other methods do.

Reply:
Checked only for nil